File: microservices/business/user-microservice/user-microservice-out-mongo-port/src/main/java/com/joserbatista/service/user/outbound/mongo/repository/UserMongoRepository.java
package com.joserbatista.service.user.outbound.mongo.repository;

import com.joserbatista.service.user.outbound.mongo.document.UserDocument;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.Optional;

public interface UserMongoRepository extends MongoRepository<UserDocument, String> {

    Optional<UserDocument> findByUsername(String username);

    boolean existsByUsername(String username);

}

File: microservices/_cloud/gateway-microservice/src/main/kotlin/com/joserbatista/cloud/gatewaymicroservice/auth/UserMongoRepository.kt
package com.joserbatista.cloud.gatewaymicroservice.auth

import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Mono

interface UserMongoRepository : ReactiveCrudRepository<UserDocument, String> {

    fun findByUsername(username: String): Mono<UserDocument>
}

